starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -1593,13 +1590,31 @@ sub check_star_catalog {
 		push @warn, sprintf "$alarm [%2d] Magnitude.  MAG or MAGMAX not defined \n",$i;		
 	    }
 	    else{
-		if (($maxmag - $mag) < $mag_faint_slot_diff){
-		    my $slot_diff = $maxmag - $mag;
-		    push @warn, sprintf "$alarm [%2d] Magnitude.  MAXMAG - MAG = %1.2f < $mag_faint_slot_diff \n",$i,$slot_diff;
+		if ($maxmag < $mag){
+		    push @warn, sprintf "$alarm [%2d] Magnitude.  MAXMAG < MAG \n",$i;
 		}
+                # These aren't captured in ACA-041 but are reasonable sanity checks for current
+                # star selection parameters.  Should be revisited if the MAXMAG cap of 11.2, the
+                # max guide star mag of 10.3, or the max acq star mag of 11.0 are changed.
+                my $max_guide_mag = 10.3;
+                my $max_acq_mag = 11.0;
+                my $max_mag_clip = 11.2;
+                # If the star is bright enough to not have a clipped MAXMAG, the difference should
+                # still be at least 1.4 mags
+                if (($mag < $maxmag - 1.5) and ($maxmag - $mag < 1.4)){
+		    push @warn, sprintf "$alarm [%2d] Magnitude.  MAXMAG - MAG < 1.4 \n",$i;
+                }
+                elsif (($type =~ /BOT|GUI/) and ($maxmag - $mag < $max_mag_clip - $max_guide_mag)){
+		    push @orange_warn, sprintf "$alarm [%2d] Magnitude.  Guide star MAXMAG - MAG < %.1f \n",$i,
+                        $max_mag_clip - $max_guide_mag;
+                }
+                elsif (($type =~ /ACQ/) and ($maxmag - $mag < $max_mag_clip - $max_acq_mag)){
+		    push @orange_warn, sprintf "$alarm [%2d] Magnitude.  Acquisition star MAXMAG - MAG < %.1f \n",$i,
+                        $max_mag_clip - $max_acq_mag;
+                }
```

Comment:
Honestly, we could go with either the first check OR the latter two as sanity checks.  None of this seems really necessary, but is also not going to show up in the output.

Reply:
Since I'll need to figure out rounding in Perl again, why 2 digits?  I would figure 1 decimal.

Reply:
I just figured that if there really was a maxmag of 11.249 we would want to know.  I'm not sure of the precision of maxmag in backstop, but thought that two decimal places is safe and 11.2049 would more likely just be numerics.  Maybe 3 decimals is also fine.

I thought round is perl is done with sprintf and that horrible string-can-be-a-number?

Reply:
In backstop, per my test report, we have 1.12031250e+01 .

And I thought sprintf did truncation but not rounding?

Reply:
My understanding is that the C-based sprintf format codes like 5.2f are rounding.

So it looks like it needs to be 1-digit in any case.  I guess that value doesn't have many bits, like a 16-bit float?  Anyway, no matter.

Reply:
Perl rounds according to SO:
https://stackoverflow.com/questions/9781774/truncate-not-round-decimal-places-in-sprintf

Reply:
I got confused about the 2 digits / 1 decimal started with 2.  Am I getting warmer?